### PR TITLE
Implement Pause/Continue Functionality for Timer (#17)

### PR DESCRIPTION
**Pull request by vijaykarthiktk**
### Video Demonstration
https://github.com/guyluz11/infinite_horizons/assets/39937989/fb91cc11-e0b9-469d-97d1-a37b5ed6aff9

### CODE REVIEW

File: lib/presentation/atoms/timer_atom.dart [hunk 1]
```diff
@@ -18,25 +20,60 @@ class TimerAtom extends StatelessWidget {
...
+                        }
+                      },
+                      child: Stack(
+                        children: [
+                          Positioned.fill(
+                            child: CircularProgressIndicator(
+                              backgroundColor: Colors.grey[300],
+                              color: value == CustomTimerState.counting
+                                  ? Colors.purpleAccent[100]!
+                                  : Colors.grey[500],
```
Comment (guyluz11) [inline lib/presentation/atoms/timer_atom.dart:50]: Colors should arive from theme

File: lib/presentation/atoms/timer_atom.dart [hunk 2]
```diff
@@ -18,25 +20,60 @@ class TimerAtom extends StatelessWidget {
...
+                              value: time.duration.inMilliseconds /
+                                  timer.inMilliseconds,
+                              strokeWidth: 22,
+                              strokeCap: StrokeCap.round,
+                            ),
+                          ),
+                          Align(
+                            alignment: FractionalOffset.center,
+                            child: Text(
```
Comment (guyluz11) [inline lib/presentation/atoms/timer_atom.dart:59]: Pleas use TextAtom

**Review by guyluz11 — CHANGES_REQUESTED** [review]
Please change pr to dev, next time also fork from dev.

And if you can add a pause symbol under the text when the timer paused it will be nice, no play symbole when the timer is working.

File: lib/presentation/atoms/timer_atom.dart (hunk 2 above)
Comment (vijaykarthiktk) [reply lib/presentation/atoms/timer_atom.dart:59]: I made the change, but, I ran into an Easy Localization error after switching to TextAtom. It says 'Localization key [59:39] not found.'

 I guess that means the text at that spot needs a key in my localization files, but I wasn't sure what to name it.

 Any advice on how to name Easy Localization keys properly? Knowing the right key for the text.

File: lib/presentation/molecules/timer_molecule.dart [hunk 3]
```diff
@@ -1,14 +1,36 @@
...
+    with SingleTickerProviderStateMixin {
+  late final CustomTimerController controller = CustomTimerController(
+    vsync: this,
+    begin: widget.duration,
+    end: Duration.zero,
+  );
+
+  @override
+  void initState() {
+    controller.start();
```
[the remark this answers is not in the document]
Comment (vijaykarthiktk) [reply lib/presentation/molecules/timer_molecule.dart:26]: yeah, no issues.

File: lib/presentation/atoms/timer_atom.dart (hunk 2 above)
Comment (guyluz11) [reply lib/presentation/atoms/timer_atom.dart:59]: It is just numbers so we can disable translation for this one using `translate: false,`

Comment (vijaykarthiktk) [reply lib/presentation/atoms/timer_atom.dart:59]: it works.

File: lib/presentation/atoms/timer_atom.dart (hunk 1 above)
Comment (vijaykarthiktk) [reply lib/presentation/atoms/timer_atom.dart:50]: https://github.com/guyluz11/infinite_horizons/assets/39937989/a3a403db-e0a2-47cc-b4ab-b6fe0101ae48

```
 backgroundColor: colorScheme.primaryContainer,
      color: value == CustomTimerState.counting
            ? colorScheme.primary
            : colorScheme.outline,
```
how about these colours?

Comment (guyluz11) [reply lib/presentation/atoms/timer_atom.dart:50]: Good, thanks

**Comment by vijaykarthiktk** [comment]
I pushed the updates, let me know if everything looks good for merging.

**Comment by vijaykarthiktk** [comment]
Forget to add pause button, now all set and ready to merge

**Comment by guyluz11** [comment]
> Forget to add pause symbol, now all set and ready to merge Simulator Screenshot - iPhone 15 Pro Max - 2024-05-06 at 08 44 45 Simulator Screenshot - iPhone 15 Pro Max - 2024-05-06 at 08 44 42

Can you change the black box to a play icon?
image

Also pr should be to dev please change it.

**Review by guyluz11 — CHANGES_REQUESTED** [review]

File: lib/presentation/atoms/timer_atom.dart [hunk 4]
```diff
@@ -18,25 +21,90 @@ class TimerAtom extends StatelessWidget {
...
+                                    horizontal: 20,
+                                  )
+                                : null,
+                            child: TextAtom(
+                              "${time.minutes}:${time.seconds}",
+                              style: TextStyle(
+                                fontSize: 33.0,
+                                color: colorScheme.onBackground,
+                                fontWeight: FontWeight.bold,
+                              ),
```
[the remark this answers is not in the document]
Comment (vijaykarthiktk) [reply lib/presentation/atoms/timer_atom.dart]: Sure.

**Comment by vijaykarthiktk** [comment]
Changed to dev.

File: lib/presentation/atoms/text_atom.dart [hunk 5]
```diff
@@ -58,5 +63,5 @@ enum TextVariant {
   regular,
   medium,
   title,
-  ;
+  time;
```
Comment (guyluz11) [inline lib/presentation/atoms/text_atom.dart:66]: use titleLarge or headlineMedium, there is no reason to add a new one

Comment (vijaykarthiktk) [reply lib/presentation/atoms/text_atom.dart]: Changed to TextVariant.title

Comment (guyluz11) [reply lib/presentation/atoms/text_atom.dart]: Thanks

**Comment by guyluz11** [comment]
@vijaykarthiktk I see there are merge conflicts, can you fix them?

File: lib/presentation/atoms/timer_atom.dart [hunk 6]
```diff
@@ -1,10 +1,13 @@
-import 'package:circular_countdown_timer/circular_countdown_timer.dart';
+import 'dart:math';
+
+import 'package:custom_timer/custom_timer.dart';
 import 'package:flutter/material.dart';
+import 'package:infinite_horizons/presentation/atoms/text_atom.dart';
```
Comment (guyluz11) [inline lib/presentation/atoms/timer_atom.dart:5]: Please import `import 'package:infinite_horizons/presentation/atoms/atoms.dart';` instead

**Review by guyluz11 — CHANGES_REQUESTED** [review]

File: lib/presentation/atoms/timer_atom.dart (hunk 6 above)
Comment (vijaykarthiktk) [reply lib/presentation/atoms/timer_atom.dart]: Updated.

**Comment by vijaykarthiktk** [comment]
> @vijaykarthiktk I see there are merge conflicts, can you fix them?

Yeah I see. Now fixed.

**Review by guyluz11 — APPROVED** [review]

**Pull request merged**